### PR TITLE
BREAKING: WithErrorHandler server option to engine option

### PR DESCRIPTION
**Pull request by dylanhitt**
Moves WithErrorHandler to an engine option.

### CODE REVIEW

File: engine_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func TestWithErrorHandler(t *testing.T) {
+	t.Run("default engine", func(t *testing.T) {
+		e := NewEngine()
+		err := NotFoundError{
+			Err: errors.New("Not Found :c"),
+		}
+		errResponse := e.ErrorHandler(err)
+		require.ErrorAs(t, errResponse, &HTTPError{})
+	})
+	t.Run("custom handler", func(t *testing.T) {
```
Comment (ccoVeille) [inline engine_test.go:19]: Please also test what happens when you try to pass nil as an ErrorHandler

Comment (dylanhitt) [reply engine_test.go:19]: I was thinking about this. I would rather just panic in the With function. I think we should do that in all cases where the input can’t be nil

Comment (ccoVeille) [reply engine_test.go:19]: That's my point, yes. Testing it allows to discuss about what should be done.

I'm not fan of panic, but yes there should be some error management.

Comment (dylanhitt) [reply engine_test.go:19]: The other option is to check nil and default to our ErrorHandler which I also don't like. The pattern to panic here has been established in previous option func in both routes and Server.

**Comment by ccoVeille** [comment]
Closed in favor of #320
